tests/system/feature_store/test_feature_validator.py

```diff
@@ -1,84 +1,217 @@
+from mlrun.feature_store.api import delete_feature_set
 from mlrun.features import Feature, Validator
 from tests.system.base import TestMLRunSystem
 
 
-# Without enterprise functions (without v3io mount and pipelines)
 @TestMLRunSystem.skip_test_if_env_not_configured
+@pytest.mark.enterprise
 class TestFeatureValidator(TestMLRunSystem):
-    def validator_types(self):
-        for validator in validators_check_type.keys():
-            for test_state, test_value in validators_check_type[validator]:
+    def test_validator_types(self):
+        for validator in self._validators_check_type.keys():
+            for test_state, test_value in self._validators_check_type[validator]:
                 feature = Feature(validator)
                 feature.validator = Validator(True, "Info")
 
                 feature.validator.set_feature(feature)
                 ok, result = feature.validator.check(test_value)
                 assert ok == test_state, f"Check type error, result:{result}"
 
+    def test_feature_set_entities(self):
+        # Test add entities with different syntax names, types and descriptions
+        # (including featureset save)
+        featureset_name = "testset_entities"
+        myset = FeatureSet(featureset_name, entities=[Entity("key1")])
+        myset["f1"] = Feature(ValueType.INT64, description="my f1")
+
+        entities = [
+            {"name": "key2", "vtype": ValueType.BOOL, "result": True},
+            {"name": "key_3", "vtype": ValueType.INT8, "desc": "abcde", "result": True},
+            {
+                "name": "key__4",
+                "vtype": ValueType.INT16,
+                "desc": "1234",
+                "result": True,
+            },
+            {
+                "name": "key 5",
+                "vtype": ValueType.INT32,
+                "desc": "+ěščřžýáíéúů",
+                "result": True,
+            },
+            {"name": "key_6", "vtype": ValueType.INT64, "result": True},
+            {"name": "key_7", "vtype": ValueType.INT128, "result": True},
+            {"name": "key_8", "vtype": ValueType.UINT8, "result": True},
+            {"name": "key__9", "vtype": ValueType.UINT16, "result": True},
+            {"name": "key 10", "vtype": ValueType.UINT32, "result": True},
+            {"name": "key_11", "vtype": ValueType.UINT64, "result": True},
+            {"name": "key_12", "vtype": ValueType.UINT128, "result": True},
+            {"name": "key_13", "vtype": ValueType.FLOAT, "result": True},
+            {"name": "key_14", "vtype": ValueType.FLOAT16, "result": True},
+            {"name": "key_15", "vtype": ValueType.DOUBLE, "result": True},
+            {"name": "key_16", "vtype": ValueType.BFLOAT16, "result": True},
+            {"name": "key_17", "vtype": ValueType.BYTES, "result": True},
+            {"name": "key_18", "vtype": ValueType.STRING, "result": True},
+            {"name": "key_19", "vtype": ValueType.DATETIME, "result": True},
+        ]
```

Comment:
please add cases with result:False here and in the test_feature_set_features

Reply:
@katyakats, I tested also "invalid" names of entities and features (long names, unexpected names, forbidden prefix such as 't_', etc. directly in jupyter via call MLRun API) which can generate issues in code later (e.g. in case of aggregation, etc.) but in current implementation the FeatureSet.Save method, everything is possible to save to DB (because validation is missing). It means, it is not possible to write code with negative results. 

I wrote these positive tests, because I would like to be sure that valid code (valid names of entities and features) will work also after future changes of source code.

I guess that negative results can be generate in case, that add_entity, add_feature or Save methods will contain validation (which is missing right now, but will be added in future - I hope).

Is it fine for you to keep only positive results? 

BTW: It is not possible to add validation without synchronization with other people from Iguazio team and I am from different company (this synchronization is not easy for me because I am little disconnected from Iguazio team :-( ).